editoast/src/modelsv2/authn.rs

```diff
@@ -0,0 +1,224 @@
+use std::{collections::HashSet, sync::Arc};
+
+use diesel::{dsl, prelude::*};
+use diesel_async::{scoped_futures::ScopedFutureExt, AsyncConnection, RunQueryDsl};
+use editoast_authz::{
+    authorizer::{StorageDriver, UserInfo},
+    roles::{BuiltinRoleSet, RoleConfig},
+};
+use editoast_models::DbConnectionPoolV2;
+use itertools::Itertools;
+
+use crate::tables::*;
+
+pub struct PgAuthDriver<B: BuiltinRoleSet + Send + Sync> {
+    pool: Arc<DbConnectionPoolV2>,
+    _role_set: std::marker::PhantomData<B>,
+}
+
+impl<B: BuiltinRoleSet + Send + Sync> PgAuthDriver<B> {
+    pub fn new(pool: Arc<DbConnectionPoolV2>) -> Self {
+        Self {
+            pool,
+            _role_set: Default::default(),
+        }
+    }
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum AuthDriverError {
+    #[error(transparent)]
+    DieselError(#[from] diesel::result::Error),
+    #[error(transparent)]
+    PoolError(#[from] editoast_models::db_connection_pool::DatabasePoolError),
+}
+
+// TODO: use load_stream instead of load to avoid useless allocations
```

Comment:
Do we really worry about performance on loading this information? I don't believe we expect an important number of objects. I would rather not leave such `TODO` that probably will never be addressed and might not even be useful, what do you think?

Reply:
Seems like a good plan.